client.go

```diff
@@ -37,8 +50,85 @@ func defaultSocketName() string {
 	return path.Join(os.TempDir(), fmt.Sprintf("emacs%d", os.Getuid()), "server")
 }
 
+// defaultEmacsDir returns the default Emacs configuration directory (aka `.emacs.d`) for the current user.
+func defaultEmacsDir() string {
+	var emacsDirName string // default folder name used by emacs
+	var emacsDir string
+
+	if runtime.GOOS == "windows" {
+		emacsDirName = ".emacs.d" // windows always uses '.emacs.d'
+	} else {
+		emacsDirName = "emacs"
+	}
+
+	xdgPathA := path.Join(os.Getenv("XDG_CONFIG_HOME"), emacsDirName)
+	xdgPathB := path.Join(os.Getenv("HOME"), ".config", emacsDirName) // user following convention without XDG_CONFIG_HOME set
+	legacyPath := path.Join(os.Getenv("HOME"), emacsDirName)          // windows (if HOME is set) and emacs pre-v27
```

Comment:
I think it could be either, but I believe it defaults to  `~/.config/emacs` if you are following the `XDG` specification (which is most linux distros now?)

Reply:
Sorry, I meant to only highlight the last line. 

It should be:

xdgPathA := path.Join(os.Getenv("XDG_CONFIG_HOME"), emacsDirName)
xdgPathB := path.Join(os.Getenv("HOME"), ".config", emacsDirName) 
legacyPath := path.Join(os.Getenv("HOME"), ".emacs.d")